compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/cfa/CfgTraverser.kt

```diff
@@ -31,14 +31,12 @@ fun ControlFlowGraph.traverse(
     traverse(direction, visitor, null)
 }
 
+// ---------------------- Path-sensitive data collection -----------------------
 
-// --------------------- Path-insensitive data collection ----------------------
-
-// TODO: Deprecate and make all existing checkers path-sensitive
 fun <I : ControlFlowInfo<I, K, V>, K : Any, V : Any> ControlFlowGraph.collectDataForNode(
     direction: TraverseDirection,
     initialInfo: I,
-    visitor: ControlFlowGraphVisitor<I, Collection<I>>
+    visitor: ControlFlowGraphVisitor<I, Collection<Pair<EdgeLabel, I>>>
```

Comment:
I'm not sure, but may be some kind of map is better than `Collection<Pair>` here?

Reply:
Hm, not really. Say we use `Map<EdgeLabel, I>`. If a certain node has multiple incoming info with the same label (and usually will do so, because the common path is labeled with `NormalPath`), now the value part, `I`, needs to be `merge`d, but that's not what the traverser should do; instead, here we literally collect them as a collection of pair, and let the `visitor` do its job: It could apply labels; merge infos; etc. with its own orders/logics.